### ISSUE TITLE
cnameAliasList set to * seems to block some content on websites

### ISSUE THREAD

**Issue by troysjanda** [uBlockOrigin/uBlock-issues#788]
### Prerequisites

- [x ] I verified that this is not a filter issue
    - Filter issues MUST be reported at [filter issue tracker](https://github.com/uBlockOrigin/uAssets/issues)
    
- [x ] This is not a support issue or a question
    - Support issues and questions are handled at [/r/uBlockOrigin](https://old.reddit.com/r/uBlockOrigin/)
    
- [ x] I performed a cursory search of the issue tracker to avoid opening a duplicate issue
    - Your issue may already be reported.
- I tried to reproduce the issue when...
    - [x ] uBlock Origin is the only extension
    - [x ] uBlock Origin with default lists/settings
    - [x ] using a new, unmodified browser profile
- [x ] I am running the latest version of uBlock Origin
- [x ] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I report is not a normal behavior

### Description

Seem with the advance setting **cnameAliasList** set to * blocks some content on Sports.yahoo.com 

Screen shots:
With **cnameAliasList** set to unset (default)
![alt text](https://i.postimg.cc/t4vFhCyw/wihout.png)
With **cnameAliasList** set to *
![alt text](https://i.postimg.cc/d1rdP2dc/with.png)

### A specific URL where the issue occurs
sports.yahoo.com, news.yahoo.com only tested  these

### Steps to Reproduce

1. set advance setting cnameAliasList to *
2. Visit sports.yahoo.com
3. [and so on...]

### Expected behavior:

web page should display corrently

### Actual behavior:
partial loading, navbar gone and most articles not showing and some images not displaying

### Your environment

* uBlock Origin version: latest dev
* Browser Name and version: FF 70.0.1
* Operating System and version: Windows 10 ENT, 1909 18363.476

**Comment by troysjanda** [comment]
I have this issue on 2 different computers not sure why its unable to reproduce

**Comment by uBlock-user** [comment]
Open the logger, put `s.yimg.com` in the filter box and refresh `sports.yahoo.com` does it appear blocked by any filterlist or by you ?

That breakage only happens if `s.yimg.com` is blocked.

**Comment by troysjanda** [comment]
Its not block or listed as blocked except this single time

https://s.yimg.com/dh/ap/ad/advertisement/sponsor/promo/ad.png

**Comment by troysjanda** [comment]
Fixed, when I set the setting to * there was a space before the * did not realize it. I removed the space and reloaded sports.yahoo.com and it loads correctly.

**Comment by gorhill** [comment]
Do not open filter issue here please, **especially** filter issues arising as a result of you altering uBO's filtering behavior by modifying advanced user settings.

**Comment by gwarser** [comment]
| | | | | | | | |
|:--- |:--- |:--- |:--- |:--- |:--- |:--- |:--- |
| +63 | -share-buttons.$script | -- | sports.yahoo.com | 3 | script | `https://edge.gycpi.b.yahoodns.net/xe/tdv2-sports-site/js/react-share-buttons.e081619b08a957ecf8ee.min.js` | cnameOf=s.yimg.com |
| +63 | @@\|\|yimg.com^*/react-share-buttons.$script,domain=yahoo.com | ++ | sports.yahoo.com | 3 | script | `https://s.yimg.com/xe/tdv2-sports-site/js/react-share-buttons.e081619b08a957ecf8ee.min.js` |

`-share-buttons` is whitelisted when comming from `yimg.com` domain, but with CNAME it comes from `yahoodns.net`

**Comment by troysjanda** [comment]
> 
> 
> Do not open filter issue here please, **especially** filter issues arising as a result of you altering uBO's filtering behavior by modifying advanced user settings. Complaining that uBO's filtering behavior is modified by setting `cnameAliasList` to `*` is equivalent to complaining that uBO's filtering behavior is modified by adding a custom dynamic filtering rule: you are entirely responsible for the result.

Mr Gorhill,

I was careful in my initial testing and had disabled all filters except the Ubo list section and the result was the same with that setting set to * some of the content would not appear. This is the reason I filed the issue. I later saw my error, when I set the advance setting from unset to * there was  a space before the * when I removed that space the issue went away. later when ublock-user tagged with invalid he had me test a filter in the logger--s.yimg.com-- and the results were it was not blocked as this time I had already restore all my filterlists. s.yimg.com was not blocked but I had a single filter that was similar s.yimg.com/dh/ap/ad/advertisement/sponsor/promo/ad.png. I hope that clear it up a bit.

Thanks

**Comment by gorhill** [comment]
> when I set the advance setting from unset to * there was a space before the * when I removed that space the issue went away

Completely unrelated, uBO trims spaces before processing the value, [here is the code](https://github.com/gorhill/uBlock/blob/3a564c199260a857f3d78d5f12b8c3f1aa85b865/platform/firefox/vapi-webrequest.js#L179) that does this.

**Comment by troysjanda** [comment]
> 
> 
> > when I set the advance setting from unset to * there was a space before the * when I removed that space the issue went away
> 
> Completely unrelated, uBO trims spaces before processing the value, [here is the code](https://github.com/gorhill/uBlock/blob/3a564c199260a857f3d78d5f12b8c3f1aa85b865/platform/firefox/vapi-webrequest.js#L179) that does this.

thank you this was my experience with this issue I can't explain other than what I have in the future ill not open issues. please have a nice day and god bless.

**Comment by gorhill** [comment]
I am merely asking for due diligence when opening issues, _especially_ when it involves advanced user features, and especially when the primary purpose of the advanced user feature being modified is to modify the filtering behavior of uBO -- it is to be expected that as a result more or less stuff may end up being blocked. The _"new issue"_ template says:

> See what the logger <https://github.com/gorhill/uBlock/wiki/The-logger> reports when you reproduce the issue, this will help you determine whether this is a filter issue.

When an issue is about undue blocking, I do expect the first step to be investigation using the logger, and this was nowhere to be found in your report to make the case uBO was blocking stuff it shouldn't have blocked.

As for `cnameAliasList`, a space before `*` is not going to make a difference. However in investigating this specific claim, I found that a space _after_ may affect the result, but this would be the equivalent of using `unset`, i.e. to not enable CNAME lookup, which is currently the default behavior.

**Comment by uBlock-user** [comment]
That setting cannot block anything, all it does is pull the CNAME data for all the hostnames and **only** shows those ones who don't match; in the popup panel and the logger, thats it. There's no blocking involved in this functionality.

**Comment by gorhill** [comment]
> and only shows those ones who don't match

It does not only show, it replays them through the filtering engine as any other network requests. Those shown in the logger are those replayed through uBO's filtering engine.

There is blocking involved when a replayed URL matches a filter/rule, which is the case for `eulerian.net` (good), or as I have observed `googleadapis.l.google.com` (CNAME for `fonts.google.com`) which is a false positive.

**Comment by uBlock-user** [comment]
> There is blocking involved when a replayed URL matches a filter/rule, which is the case for eulerian.net (good),

Yes but this setting is not directly responsible for blocking, thats my point.

**Comment by troysjanda** [comment]
> I do expect the first step to be investigation using the logger, and this was nowhere to be found in your report to make the case uBO was blocking stuff it shouldn't have blocked.

I did use the logger and reported that I disabled all filters except Ubo section and the issue persisted. The only thing that I did to restore the missing elements was reset the cnameAliasList to unset and refreshed the page and it was normal. 

This would have made since that the trailing space in fact made cnameAliasList disabled (default) thus making sports.yahoo.com display correctly. I purposefully tested this with all lists disabled except Ubo section lists. This led me to believe that it was not a FILTER issue.

Maybe staring/testing code all day at work fried my brain. The conclusion that I made were, it was not a FILTER issue, since all filters were disabled. Seemed to be caused by enabling cnameAliasList, and restored once reset to default.
